### ISSUE TITLE
Wrap() with struct (not pointer) fails silently

### ISSUE THREAD

**Issue by teleclimber** [mfcochauxlaberge/jsonapi#117]
Passing a struct to `Wrap` should work, according to the docs, however it fails silently. It accepts the struct and returns a value but getting any attributes from the wrapped value returns its zero-value (or something along those lines).

To see this, in the `wrapper_test.go` test `TestWrapper` change this line: 

	wrap1 := Wrap(res1)

to 

	wrap1 := Wrap(*res1)

The test fails, though it should still pass. There are many errors:

```
--- FAIL: TestWrapper (0.00s)
    /Documents/Code/jsonapi/wrapper_test.go:67: 
        	Error Trace:	wrapper_test.go:67
        	Error:      	Not equal: 
        	            	expected: "res123"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-res123
        	            	+
        	Test:       	TestWrapper
        	Messages:   	id
    /Documents/Code/jsonapi/wrapper_test.go:71: 
        	Error Trace:	wrapper_test.go:71
        	Error:      	Not equal: 
        	            	expected: "res123"
        	            	actual  : "another-id"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-res123
        	            	+another-id
        	Test:       	TestWrapper
        	Messages:   	set id

//.... many more errors...
```

**Comment by mfcochauxlaberge** [comment]
#118 should fix it. Let me know if that works for you.

**Comment by teleclimber** [comment]
I still think there is something wrong with wrapped structs (non-pointer). They return `""` for most gets.

Take the test you added and change it to be a non-pointer, and just assert that what you put in is what you get out:

```
func TestWrapStruct(t *testing.T) {
	assert := assert.New(t)

	res1 := mockType1{
		ID:  "res123",
		Str: "a_string",
	}

	wrap1 := Wrap(res1)

	// ID, type, field
	id, typ := wrap1.IDAndType()
	assert.Equal(res1.ID, id, "id")
	assert.Equal("mocktypes1", typ, "type")
	assert.Equal(res1.Str, wrap1.Get("str"), "str field")
}
```

This should pass, if I understand the idea of wrappers correctly, but I get this:

```
--- FAIL: TestWrapStruct (0.00s)
    /Code/jsonapi/wrapper_test.go:53: 
        	Error Trace:	wrapper_test.go:53
        	Error:      	Not equal: 
        	            	expected: "res123"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-res123
        	            	+
        	Test:       	TestWrapStruct
        	Messages:   	id
    /Code/jsonapi/wrapper_test.go:55: 
        	Error Trace:	wrapper_test.go:55
        	Error:      	Not equal: 
        	            	expected: "a_string"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-a_string
        	            	+
        	Test:       	TestWrapStruct
        	Messages:   	str field
FAIL
FAIL	github.com/mfcochauxlaberge/jsonapi	0.013s
FAIL
```

**Comment by mfcochauxlaberge** [comment]
Wow, I tried to fix this at midnight and realized I my PR's pipeline had not even passed and res1 was still a pointer.

That being said, I couldn't reproduce the error you just mentioned exactly. I pushed a different fix. Now the pipeline passes. Let me know if that works.

**Comment by teleclimber** [comment]
Ah right. I should have pulled the PR before doing my little test, duh.

In any case if `TestWrapStruct` passes as it's currently written, then that solves the problem I experienced.

**Comment by mfcochauxlaberge** [comment]
I merged #118 Should be good now.
